PORTING.md

```diff
@@ -0,0 +1,327 @@
+```
+
+And correct CMSIS device header and device macro:
+```xml
+<header>sam4.h</header>
+<define>SAM4LC4B</define>
+```
+
+You need to implement at least three drivers: `core`, `clock` and `gpio`.
+
+Make sure you give the `core` driver the right information about the memory map. This information is used to generate the linkerscript.
+Don't worry about different memories right now, just add the main FLASH and one SRAM region here (called `sram1`).
+```xml
+<driver type="core" name="cortex">
+  <memory access="rx" start="0x8000000" name="flash" size="64"/>
+  <memory access="rwx" start="0x20000000" name="sram1" size="12"/>
+```
+
+Add all or part of the vector table. No need to add the Cortex-M exceptions, we do that for you.
+The `name` will be appended with `_Handler`, so declaring `extern "C" void WWDG_Handler(void) {...}` anywhere will link that function into the vector table (note the C linkage!).
+```xml
+  <vector position="0" name="WWDG"/>
+  <vector position="1" name="PVD"/>
+  <!-- ... -->
+  <vector position="81" name="FPU"/>
+</driver>
+```
+
+The `clock` driver just needs to be present, we'll hardcode the specifics in code:
+```xml
+<driver type="clock" name="sam4"/>
+```
+
+The `gpio` driver should contain some GPIOs, just add the ones you need. Adding more later is cheap. Ignore the alternate functions for now, you only want to blink an LED.
+```xml
+<driver type="gpio" name="sam4">
+  <gpio port="A" id="0"/>
+  <gpio port="A" id="1"/>
+<driver>
+```
+
+Remove all other drivers.
+
+### Making the build system aware
+
+Our build system is organically grown, er, I mean, *has* organically grown. Therefore it is a bit hacky in places and you now need to make it aware of the new devices from a new vendor:
+
+1. [Split the device identifier into the vendor naming scheme useful](https://github.com/roboterclubaachen/xpcc/blob/7b32f74cc43c7bfaf815287902ef2bd4a6baa84d/tools/device_files/device_identifier.py?ts=4#L86-L95).
+2. [Find the device file using the naming scheme](https://github.com/roboterclubaachen/xpcc/blob/3c7cd31e5aad66e47d073dc445551a21416d21eb/scons/site_tools/platform_tools.py?ts=4#L92-L103).
+3. [Thou Shalt ~~Not~~ Pass](https://github.com/roboterclubaachen/xpcc/blob/7b32f74cc43c7bfaf815287902ef2bd4a6baa84d/tools/device_files/device.py#L134-L139).
+4. [Add some platform tests for the new devices](https://github.com/roboterclubaachen/xpcc/blob/3c7cd31e5aad66e47d073dc445551a21416d21eb/scons/site_tools/platform_tools.py?ts=4#L385-L414).
+
+### Adapt the startup- and linkerscripts
+
+The [Cortex-M startup script](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/driver/core/cortex/startup.c.in?ts=4) is pretty universal. These are the steps it does:
+
+1. Generate the exception table and places it at the beginning of Flash.
+2. Fill the stack with magic values to catch unintentional stack execution.
+3. Call platform specific code (enabling internal memories, **needs porting**).
```

Comment:
> I am also a bit annoyed that the linkerscripts are in the cortex/stm32/ folder, instead of the cortex/linkerscript folder ;-P

what prevents you from moving the file?